### PR TITLE
Create Deck Settings Page

### PR DESCRIPTION
**Pull request by hidesminimally**
image

Creates the deck settings page 

Also currently this is no longer working on firefox. Haven't had a chance to look into it...

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -71,19 +71,20 @@ module.exports = {
     'prefer-template': 'error',
 
     'react-hooks/rules-of-hooks': 'error',
-    'react-hooks/exhaustive-deps': 'warn',
 
     // Place to specify ESLint rules. Can be used to overwrite rules
     // specified from the extended configs
 
+    'react-hooks/exhaustive-deps': 'off',
```
Comment (hidesminimally) [inline .eslintrc.js:78]: This was an overreaction to a slight annoyance, am open to turning this back to warning. What do you think @lbkchen

File: src/db/db.ts [hunk 2]
```diff
@@ -20,15 +39,16 @@ class DB {
...
-    localforage.config({
-      driver: [
-        localforage.INDEXEDDB,
-        localforage.LOCALSTORAGE,
-        localforage.WEBSQL,
-      ],
-      name: DB_NAME,
-      version: 1.0,
-    });
+    // Commenting this out cause it leads to serving crashing consistently
```
Comment (hidesminimally) [inline src/db/db.ts:42]: Not sure what's going on here, but it was needed to run the server

File: src/db/db.ts [hunk 3]
```diff
@@ -38,12 +58,95 @@ class DB {
...
+      name: DB_NAME,
+      storeName: ANKI_STORE_NAME,
+    });
 
     if (process.env.NODE_ENV === 'development') {
       console.log('INITIALIZED DB');
     }
   }
 
+  async getSql() {
```
Comment (hidesminimally) [inline src/db/db.ts:71]: Not sure if all this SQL stuff belongs here, maybe can put it in alternative file just like we do for db and initialize a singleton. Definitely smells like it's coupling things that should not be coupled.

File: .eslintrc.js (hunk 1 above)
Comment (lbkchen) [reply .eslintrc.js:78]: ya this one can definitely get noisy. I find that it does occasionally catch subtle bugs when adding dependencies to `useEffect` and the like, but also it tends to be too exhaustive sometimes and unaware of expected code context/behavior

File: src/db/db.ts (hunk 2 above)
Comment (lbkchen) [reply src/db/db.ts:42]: oh wow interesting, so it still works without these lines? :/ weird

File: src/db/db.ts (hunk 3 above)
Comment (lbkchen) [reply src/db/db.ts:71]: I think I agree w/ur intuition here

File: src/db/db.ts [hunk 4]
```diff
@@ -130,6 +233,7 @@ class DB {
   }
 }
 
+// When does this get initialized?
```
Comment (lbkchen) [inline src/db/db.ts:236]: good question, I think it gets initialized once for the whole bundle, depending on which file imports it first, but we should test that w/some log statements if it's important

Comment (hidesminimally) [reply src/db/db.ts:236]: sounds similar to what i thought as well. not important, i was just mainly curious

File: src/db/db.ts (hunk 3 above)
Comment (hidesminimally) [reply src/db/db.ts:71]: sounds good i'll get to this sometime

File: .eslintrc.js (hunk 1 above)
Comment (hidesminimally) [reply .eslintrc.js:78]: ok for now i'll leave it cause doesn't sound like strong opinions from either of us

**Pull request merged**